modules/biobank/php/optionsendpoint.class.inc

```diff
@@ -0,0 +1,166 @@
+<?php declare(strict_types=1);
+/**
+ * This implements the Options endpoint class for the Biobank Module.
+ *
+ * PHP Version 7.2
+ *
+ * @category   API
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris
+ */
+namespace LORIS\biobank;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * A class for handling the /biobank/options/ endpoint.
+ *
+ * PHP Version 7.2
+ *
+ * @category   API
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris/
+ */
+class OptionsEndpoint implements RequestHandlerInterface
+{
+    /**
+     * Returns true if user has access to this endpoint.
+     *
+     * @param \User $user The user whose access is being checked.
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return true;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return ['GET'];
+    }
+
+    /**                                                                         
+     * This function passes the request to the handler. This is necessary since 
+     * the Endpoint bypass the Module class.                                    
+     *                                                                          
+     * XXX: This function should be extracted to a parent class.                   
+     *                                                                          
+     * @param ServerRequestInterface  $request The PSR7 request.                
+     * @param RequestHandlerInterface $handler The request handler.             
+     *                                                                          
+     * @return ResponseInterface The outgoing PSR7 response.                    
+     */  
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ) : ResponseInterface {
+        return $handler->handle($request);
+    }
+
+    /**
+     * Handles endpoint requests.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db   = \Database::singleton();
+        $user = $request->getAttribute('user');
+
+        try {
+            switch($request->getMethod()) {
+            case 'GET':
```

Comment:
Should this be doing GET and OPTIONS? If so this code should match the array above

Reply:
The valid request methods for this class is only `GET`. An `OPTIONS` request is simply sent when determining what methods are allowed.

Reply:
@xlecours could you provide some clarity as to whether this is done correctly?

Reply:
We should add OPTIONS to the allowed methods where it is implemented. I'll do it on #4244

Reply:
done.